internal/engine/experiment/tlsmiddlebox/tlsmiddlebox.go

```diff
@@ -0,0 +1,179 @@
+package tlsmiddlebox
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/ooni/probe-cli/v3/internal/engine/experiment/tlsmiddlebox/internal"
+	"github.com/ooni/probe-cli/v3/internal/model"
+)
+
+const (
+	testName    = "tlsmiddlebox"
+	testVersion = "0.1.0"
+)
+
+// Config contains the experiment configuration.
+type Config struct {
+	// ResolverURL is the default DoH resolver
+	ResolverURL string `ooni:"URL for DoH resolver"`
+
+	// SNIPass is the SNI value we don't expect to be blocked
+	SNIPass string `ooni:"the SNI value to cal"`
+
+	// Delay is the delay between each iteration (in milliseconds).
+	Delay int64 `ooni:"delay between consecutive iterations"`
+
+	// Iterations is the default number of interations we trace
+	Iterations int `ooni:"iterations is the number of iterations"`
+
+	// SNI is the SNI value to use.
+	SNI string `ooni:"the SNI value to use"`
+}
+
+func (c Config) resolverURL() string {
+	if c.ResolverURL != "" {
+		return c.ResolverURL
+	}
+	return "https://mozilla.cloudflare-dns.com/dns-query"
+}
+
+func (c Config) snipass() string {
+	if c.SNIPass != "" {
+		return c.SNIPass
+	}
+	return "google.com"
+}
+
+func (c Config) delay() time.Duration {
+	if c.Delay > 0 {
+		return time.Duration(c.Delay) * time.Millisecond
+	}
+	return 100 * time.Millisecond
+}
+
+func (c Config) iterations() int {
+	if c.Iterations > 0 {
+		return c.Iterations
+	}
+	return 20
+}
+
+func (c Config) sni(address string) string {
+	if c.SNI != "" {
+		return c.SNI
+	}
+	return address
+}
+
+// TestKeys contains the experiment results.
+type TestKeys struct {
+	DNSLookUp  *model.ArchivalDNSLookupResult    `json:"dns_lookup"`
```

Comment:
Small nitpick, the convention we adopt in the base data format is to put DNS queries inside of the `queries` key (see: https://github.com/ooni/spec/blob/master/data-formats/df-002-dnst.md)

Reply:
Thanks for pointing this out!